source/s3_meta_request.c

```diff
@@ -244,14 +245,7 @@ void aws_s3_default_signing_config(
     signing_config->signed_body_value = g_aws_signed_body_value_unsigned_payload;
 }
 
-static void s_s3_meta_request_start_destroy(void *user_data) {
```

Comment:
We weren't actually using "internal references" (references that don't block clean-up from initiating, but prevent it from finishing) for anything in the meta request.  This meant that clean up was already immediate, so just cutting out the middle-man here.